circuit_knitting/cutting/cutting_evaluation.py

```diff
@@ -238,12 +239,56 @@ def _append_measurement_circuit(
     return qc
 
 
-def _generate_cutting_experiments(
+def generate_cutting_experiments(
     circuits: QuantumCircuit | dict[str | int, QuantumCircuit],
     observables: PauliList | dict[str | int, PauliList],
-    num_samples: int,
-) -> tuple[list[list[list[QuantumCircuit]]], list[tuple[Any, WeightType]], list[float]]:
-    """Generate all the experiments to run on the backend and their associated coefficients."""
+    num_samples: int | float,
+) -> tuple[
+    list[QuantumCircuit] | dict[str | int, list[QuantumCircuit]],
+    list[tuple[float, WeightType]],
+    list[list[list[QuantumCircuit]]],
+]:
```

Comment:
Eventually, we want this to return only the first two elements, right?  (not `subexperiments_legacy`).

I think it would be pretty straightforward to modify `execute_experiments` to work with `subexperiments_out` instead of `subexperiments_legacy`, wouldn't it?  It would, of course, mean changing a data type in `CuttingExperimentResults` and then also dealing with that changed type in `reconstruct_expectation_values`.  (Without looking, I'm guessing this may be where you took things on your `sampler-out` branch?)

But without making this PR take on too much scope, I suggest the following:

- Restore the leading underscore, to make this modified `_generate_cutting_experiments` (one which returns a 3-tuple) be private.
- Move your new docstring to a `generate_cutting_experiments` right above this one.  After the docstring, this function should simply call `_generate_cutting_experiments`, throw away `subexperiments_legacy`, and then return.
- `execute_experiments` can also call `_generate_cutting_experiments`

This way, the interface exposed by `generate_cutting_experiments` will be stable over time.

---

CODA: Considering my follow-up comment about `weights` below, we might have to break this function's interface eventually, anyway -- ideally before v0.4 but we might not have enough time.  It might be nice to do a little bit more thinking about whether we should break this sooner than later (I'll have to think about it over the next few hours, and I don't want to delay giving some feedback while I do).  At the moment, I'm leaning toward just biting the bullet and changing `weights`, even if we don't have time to finish #262, but I want to think more critically about it.

Reply:
I've also been thinking it may actually be possible to support a deprecated workflow for a while without an unreasonable amount of effort. Let's see how it shakes out in these upcoming PRs